### PR TITLE
Fixed SAIC-389: added new action to check filtered entries

### PR DESCRIPTION
**Pull request by realsystem**

### CODE REVIEW

File: cloudferrylib/os/actions/check_filter.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        image_resource = self.cloud.resources[utl.IMAGE_RESOURCE]
+        if search_opts_img and search_opts_img.get('images_list'):
+            images_list = search_opts_img['images_list']
+            for img_id in images_list:
+                LOG.debug('Filtered image id: {}'.format(img_id))
+                try:
+                    img = image_resource.glance_client.images.get(img_id)
+                    if img:
+                        LOG.debug('Image is OK')
+                except Exception as e:
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: I'd suggest catching `NotFound` error only, catching all exceptions is considered bad practice in general.

File: cloudferrylib/os/actions/check_filter.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        ident_resource = self.cloud.resources[utl.IDENTITY_RESOURCE]
+        if search_opts_tenant and search_opts_tenant.get('tenant_id'):
+            tenants = search_opts_tenant['tenant_id']
+            for tenant_id in tenants:
+                LOG.debug('Filtered tenant id: {}'.format(tenant_id))
+                try:
+                    tenant = ident_resource.keystone_client.tenants.find(id=tenant_id)
+                    if tenant:
+                        LOG.debug('Tenant is OK')
+                except Exception as e:
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Same as above

File: cloudferrylib/os/actions/check_filter.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        compute_resource = self.cloud.resources[utl.COMPUTE_RESOURCE]
+        if search_opts and search_opts.get('id'):
+            instances = search_opts['id']
+            for instance_id in instances:
+                LOG.debug('Filtered instance id: {}'.format(instance_id))
+                try:
+                    instance = compute_resource.nova_client.servers.get(instance_id)
+                    if instance:
+                        LOG.debug('Instance is OK')
+                except Exception as e:
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Ditto

File: cloudferrylib/os/actions/check_filter.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        search_opts_tenant = kwargs.get('search_opts_tenant', {})
+        image_resource = self.cloud.resources[utl.IMAGE_RESOURCE]
+        if search_opts_img and search_opts_img.get('images_list'):
+            images_list = search_opts_img['images_list']
+            for img_id in images_list:
+                LOG.debug('Filtered image id: {}'.format(img_id))
+                try:
+                    img = image_resource.glance_client.images.get(img_id)
+                    if img:
+                        LOG.debug('Image is OK')
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Please provide more meaningful message, for example: "Filter config check: Image ID %s is OK"

File: cloudferrylib/os/actions/check_filter.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                except Exception as e:
+                    LOG.error('Image get error')
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Please provide more meaningful message, for example, "Filter config check: Image ID '%s' is not present in source cloud, please update your filter config. Aborting."

File: cloudferrylib/os/actions/check_filter.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                    raise e
+        ident_resource = self.cloud.resources[utl.IDENTITY_RESOURCE]
+        if search_opts_tenant and search_opts_tenant.get('tenant_id'):
+            tenants = search_opts_tenant['tenant_id']
+            for tenant_id in tenants:
+                LOG.debug('Filtered tenant id: {}'.format(tenant_id))
+                try:
+                    tenant = ident_resource.keystone_client.tenants.find(id=tenant_id)
+                    if tenant:
+                        LOG.debug('Tenant is OK')
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Same as above

File: cloudferrylib/os/actions/check_filter.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                except Exception as e:
+                    LOG.error('Tenant get error')
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Same as above

File: cloudferrylib/os/actions/check_filter.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                    raise e
+        compute_resource = self.cloud.resources[utl.COMPUTE_RESOURCE]
+        if search_opts and search_opts.get('id'):
+            instances = search_opts['id']
+            for instance_id in instances:
+                LOG.debug('Filtered instance id: {}'.format(instance_id))
+                try:
+                    instance = compute_resource.nova_client.servers.get(instance_id)
+                    if instance:
+                        LOG.debug('Instance is OK')
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Same as above

File: cloudferrylib/os/actions/check_filter.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+                except Exception as e:
+                    LOG.error('Instance get error')
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Same as above

**Comment by realsystem** [comment]
-1, need to fix func tests

**Comment by realsystem** [comment]
+1, ready for review and merge

**Comment by roman-verchikov** [comment]
+1, functional tests succeeded, can be merged.

**Pull request merged**